Doc/.ruff.toml

```diff
@@ -0,0 +1,46 @@
+target-version = "py312"  # align with the version in oldest_supported_sphinx
+fix = true
+output-format = "full"
+extend-exclude = [
+    "includes/*",
+    "tools/extensions/c_annotations.py",
+    "tools/extensions/escape4chm.py",
+    "tools/extensions/patchlevel.py",
+    "tools/extensions/pyspecific.py",
+]
```

Comment:
What's the reason for excluding these? Is the aim to get rid of this exclude list in due course, or do we intend to keep them excluded indefinitely? Maybe add a comment?

Reply:
I don't intend to touch ``includes`` for the moment, but I hope to remove the ``tools`` exclusions in due course. Shall add a comment.